### DISCUSSION TITLE
Block outgoing queries for TLD

### DISCUSSION THREAD

**Discussion by joshqou** [AdguardTeam/AdGuardHome#4816] [Q&A] [answered]
Hi. I'm using AdGuardHome to manage intranet domains and I want to prevent outgoing lookups for the TLD I'm using. Using a subdomain match on the TLD doesn't seem to work, even with $important.

Adding `[tld]0.0.0.0` to my upstream list *does* work, however agh doesn't log when it fails to resolve a query. This solution also makes invalid queries take several seconds to time out.

I want to avoid resorting to using subdomain matches for all my dnsrewrite rules as it would result in my TXT and SRV records bleeding down to other subdomains.

**Comment by ainar-g** [comment, marked as answer]
Have you tried adding:

```none
||tld^
```

to your *Custom filtering rules?*

**Reply by joshqou** [reply]
I previously tried that but with `$dnsrewrite=NXDOMAIN` to set the proper answer code, however that resulted in it overwriting everything in `tld`. This is a better solution, however agh blocks unmatched domains with the set blocking mode, and I can't set the proper answer code without overwriting my internal domain rules. Changing the blocking mode to MXDOMAIN could also break some user filters from the setting description.
